### PR TITLE
fix: Use custom thread factor class in SolverManager

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**

### CODE REVIEW

File: core/src/test/java/ai/timefold/solver/core/api/solver/SolverManagerTest.java [hunk 1]
```diff
@@ -1010,4 +1011,36 @@ void terminateScheduledSolverJobEarly_returnsInputProblem() throws ExecutionExce
...
+
+        @Override
+        public Thread newThread(Runnable runnable) {
+            return new Thread(runnable, CUSTOM_THREAD_NAME);
+        }
+    }
+
+    @Test
+    @Timeout(60)
+    void threadFactoryIsUsed() throws ExecutionException, InterruptedException {
```
Comment (zepfred) [inline core/src/test/java/ai/timefold/solver/core/api/solver/SolverManagerTest.java:1026]: Nice!

**Review by zepfred — APPROVED** [review]

**Comment by triceo** [comment]
Why is this being done? Do we have a request somewhere, a justification?
This is highly unusual.

**Comment by triceo** [comment]
If this is something for the Python solver, I'd prefer if we had a code branch specifically for the Python solver, and leave the original behavior of the Java solver unchanged.

**Comment by Christopher-Chianelli** [comment]
Two reasons:

- It a genuine bug: `threadFactoryClass` on `SolverManager` literally **did nothing** before when used on `SolverManager`
- Due to how Python/JVM interacts, Python Solver requires daemon threads to terminate when used with Control-C/interrupts.

**Comment by Christopher-Chianelli** [comment]
(If thread factory unset in the `SolverManagerConfig`, original behaviour is unchanged; `Executors.newFixedThreadPool(parallelSolverCount, Executors.defaultThreadFactory()) == Executors.newFixedThreadPool(parallelSolverCount)`

**Comment by zepfred** [comment]
@triceo, this change does not affect the default behavior, which already sets the thread factory to `Executors.defaultThreadFactory()` when calling `Executors.newFixedThreadPool(parallelSolverCount);`.

**Pull request merged**